stdlib/universe/map.go

```diff
@@ -314,12 +314,16 @@ func groupKeyForObject(i int, cr flux.ColReader, obj values.Object, on map[strin
 		if !on[c.Label] {
 			continue
 		}
-		cols = append(cols, c)
 		v, ok := obj.Get(c.Label)
 		if ok {
 			vs = append(vs, v)
+			cols = append(cols, flux.ColMeta{
+				Label: c.Label,
+				Type:  flux.ColumnType(v.Type()),
+			})
 		} else {
 			vs = append(vs, execute.ValueForRow(cr, i, j))
+			cols = append(cols, c)
```

Comment:
My understanding is that this branch is dead code, unless `mergeKey` is true.  `on` would not contain the label otherwise.  Is that consistent with your understanding?

Reply:
Wait, in the `mergeKey` case, where the map `fn` is dropping that column, wouldn't we want to omit that column from the group key as well?

Reply:
Yes that is my understanding. When this function is called, a map of the group key columns are passed in. This is the `on` parameter. My understanding is that only in the case of `mergeKey=true` is it possible that one of the group key columns is not a label on `obj`. However while that label is missing from `obj`, it will be included in the output, so we want to include it in the group key.

Reply:
Right, that makes sense.